### PR TITLE
tools: build propel_windows with package.js

### PR DESCRIPTION
**Pull request by piscisaureus**
Also includes various other fixes to make package.js work on windows,
as well as some stylistic clean-ups.

### CODE REVIEW

File: tools/package.js [hunk 1]
```diff
@@ -35,55 +34,60 @@ function createPackageJson(src, dst, packageJson = {}) {
...
-                    distDir + "/tensorflow-binding.node");
-    fs.copyFileSync("build/Release/libtensorflow.so",
-                    distDir + "/libtensorflow.so");
-    fs.copyFileSync("build/Release/libtensorflow_framework.so",
-                    distDir + "/libtensorflow_framework.so");
+                    distDir + "/tensorflow-binding.node"
+    );
+    if (process.platform === "win32") {
+      fs.copyFileSync("build/Release/tensorflow.dll",
+                      distDir + "/tensorflow.dll");
```
Comment (ry) [inline tools/package.js]: Does it work without libtensorflow_framework.so?

File: tools/package.js [hunk 2]
```diff
@@ -98,8 +102,8 @@ function buildAndTest() {
                     path.join(testDir, "package.json"));
 
   process.chdir(testDir);
-  run.sh("npm install " + propelPkgFn);
-  run.sh("npm install " + tfPkgFn);
+  execSync("npm install " + propelPkgFn, { stdio: "inherit" });
+  execSync("npm install " + tfPkgFn, { stdio: "inherit" });
```
Comment (ry) [inline tools/package.js]: What's this change about?

File: tools/package.js (hunk 1 above)
Comment (piscisaureus) [reply tools/package.js]: That doesn't exist on windows. All you need is tensorflow.dll.

File: tools/package.js (hunk 2 above)
Comment (piscisaureus) [reply tools/package.js]: npm is a .bat file.
spawnSync cannot spawn bat files.

Comment (ry) [reply tools/package.js]: k

**Pull request merged**